pkg/util/objectwatcher/objectwatcher.go

```diff
@@ -73,28 +73,20 @@ func (o *objectWatcherImpl) Create(clusterName string, desireObj *unstructured.U
 		// - 1. In a reconcile process, the execution controller successfully applied resource to member cluster but failed to update the work conditions(Applied=True),
 		//   when reconcile again, the controller will try to apply(by create) the resource again.
 		// - 2. The resource already exist in the member cluster but it's not created by karmada.
-		if apierrors.IsAlreadyExists(err) {
+		//  The 'IsInvalid' conflict will happen when a resource is created if the resource kind is 'Service'
```

Comment:
```suggestion
		// The 'IsInvalid' conflict will happen when a resource is created if the resource kind is 'Service'
```

Reply:
Can you provide the reproduce steps for the `Invalid` error?

Reply:
+1

When I recreate a Service same as before created, the err info is:

```
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "error when creating \"nginx-svc.yaml\": services \"my-nginx\" already exists",
  "reason": "AlreadyExists",
  "details": {
    "name": "my-nginx",
    "kind": "services"
  },
  "code": 409
```

Reply:
reproduce  https://github.com/karmada-io/karmada/issues/1942

Reply:
I think there may be a situation that can't be handled properly: when a `NodePort` type Service is created and an allocated nodePort number is specified.

Reply:
> when a `NodePort` type Service is created and an allocated nodePort number is specified.

Yes,it is right,invalid err only happens when nodePort number is specified. `karmadactl promote service` will create a service that nodeport number is specified in member cluster,so I ignored it before.I will fix it soon.